cmd/qos.go

```diff
@@ -59,9 +55,10 @@ func getServiceQoSInstances(
 				},
 				Logger: logger,
 			}
-			if _, ok := gatewayServiceIDsIdx[serviceID]; ok {
-				gatewayQoSService[serviceID] = evm.NewServiceQoS(evmEndpointStore, logger)
-			}
+			gatewayQoSService[serviceID] = evm.NewServiceQoS(evmEndpointStore, logger)
```

Comment:
This PR is dealing with multiple aspects, and we need to focus on a single one in each PR to make it easier for any future readers to follow;
- Initialization of QoS instances for their corresponding recognized service IDs: this PR should not change anything here.
- Desired behavior re NoOp QoS, i.e. service IDs that are valid from the perspective of protocol, but have not been configured, specified, embedded, etc.: this seems to be the main target of the PR, let's focus on it exclusively.

Reply:
> This PR is dealing with multiple aspects, and we need to focus on a single one in each PR to make it easier for any future readers to follow;
> 
> * Initialization of QoS instances for their corresponding recognized service IDs: this PR should not change anything here.
> * Desired behavior re NoOp QoS, i.e. service IDs that are valid from the perspective of protocol, but have not been configured, specified, embedded, etc.: this seems to be the main target of the PR, let's focus on it exclusively.

IMO these two issues are strongly related: if we remove the limitation that a service must be specified in the config YAML **and** `service_alias.go` (now `service_qos.go`) then there is no logical reason to force an operator to configure the `services` field in the YAML when we already hardcode the mapping of `service ID -> QoS` and can derive the `gatewayQoSService` map from there.

Forcing an operator to specify this yaml field adds an unnecessary layer of configuration and mental overhead for no clear benefit. 

If the argument is just that it should be split into two separate PRs I can understand that but this work is already done and functioning in this PR; I can split it if you like but currently this functionality is blocking multiple potential gateway operators from using PATH so I'm a bit hesitant to split already completed work into multiple parts.